hydrosdk/application.py

```diff
@@ -33,250 +21,262 @@ class ApplicationStatus(Enum):
 
 
 class Application:
-    """
-    An application is a publicly available endpoint to reach your models (https://hydrosphere.io/serving-docs/latest/overview/concepts.html#applications)
-    """
+    _BASE_URL = "/api/v2/application"
 
-    def __init__(self, name, execution_graph, kafka_streaming, metadata, status, signature: ModelSignature,
-                 cluster: Cluster):
-        self.name = name
-        self.execution_graph = execution_graph
-        self.kafka_streaming = kafka_streaming
-        self.metadata = metadata
-        self.status = status
-        self.signature = signature
-        self.cluster = cluster
+    @staticmethod
+    def list_all(cluster) -> List['Application']:
+        resp = cluster.request("GET", Application._BASE_URL)
+        if resp.ok:
+            applications = [Application.from_json(cluster=cluster, application_json=app_json) for app_json in resp.json()]
+            return applications
 
-    def predictor(self, return_type=PredictorDT.DICT_NP_ARRAY) -> PredictServiceClient:
-        self.impl = MonitorableImplementation(channel=self.cluster.channel, target=self.name)
-        self.predictor_return_type = return_type
+        raise Exception(
```

Comment:
```suggestion
        raise BadResponse(
```

or something like that